CODEC/demux_CODEC.wdl

```diff
@@ -0,0 +1,277 @@
+        Array[File] fastq1
+        Array[File] fastq2
+        Array[String] batch_ids
+        Int num_parallel
+        Array[File] sample_sheet
+        String workspace
+    }
+    scatter (batch_index in range(length(batch_ids))) {
+        call SplitFastq1 {
+            input:
+                batch_id =batch_ids[batch_index],
+                fastq_read1 = fastq1[batch_index],
+                nsplit = num_parallel
+        }
+        call SplitFastq2 {
+            input:
+                batch_id =batch_ids[batch_index],
+                fastq_read2 = fastq2[batch_index],
+                nsplit = num_parallel
+        }
+        scatter (split_index in range(num_parallel)) {
+            call Demux {
+                input:
+                    paired_fastqs = (SplitFastq1.split_read1[split_index], SplitFastq2.split_read2[split_index]),
+                    sample_sheet = sample_sheet[batch_index],
+                    batch_id = batch_ids[batch_index],
+                    split = split_index
+        }
+    }
+    }
+    Array[File] demultiplexed_fastq1_all_lanes = flatten(flatten(Demux.demux_read1))
+    Array[File] demultiplexed_fastq2_all_lanes = flatten(flatten(Demux.demux_read2))
+    
+    call GroupFastqFiles as GroupFastq1 {
+      input:
+      fastq_files = demultiplexed_fastq1_all_lanes
+    }
+    call GroupFastqFiles as GroupFastq2 {
+      input:
+      fastq_files = demultiplexed_fastq2_all_lanes
+    }
+    scatter (txt_file in GroupFastq1.grouped_fastq_lists) {
+        call concatenate_and_compress_fastq as concatenate_and_compress_fastq1 {
+            input:
+                grouped_fastq_list = txt_file,
+                output_file = basename(txt_file, ".txt") + ".1.gz"
+        }
+    }
+
+    scatter (txt_file in GroupFastq2.grouped_fastq_lists) {
+        call concatenate_and_compress_fastq as concatenate_and_compress_fastq2 {
+            input:
+                grouped_fastq_list = txt_file,
+                output_file = basename(txt_file, ".txt") + ".2.gz"
+        }
+    }
+    
+    call UploadDataTable {
+        input:
+           fastq1_files = concatenate_and_compress_fastq1.output_fastq_gz,
+           fastq2_files = concatenate_and_compress_fastq2.output_fastq_gz,
+           workspace = workspace
+
+    } 
+
+    output {
+        Array[File] merged_fastq1_per_sample = concatenate_and_compress_fastq1.output_fastq_gz
+        Array[File] merged_fastq2_per_sample = concatenate_and_compress_fastq2.output_fastq_gz
+        File data_table = UploadDataTable.data_table
+    }
+}
+
+
+task SplitFastq1 {
+    input {
+        File fastq_read1
+        Int nsplit
+        String batch_id
+        Int memory = 128
+        Int disk_size = 1024
+ 
+    }
+
+    command <<<
+        set -e
+        
+        zcat ~{fastq_read1} | /CODECsuite/snakemake/script/fastqsplit.pl ~{batch_id}_split_r1 ~{nsplit}
+
+    >>>
+
+    output {
+        Array[File] split_read1 = glob("~{batch_id}_split_r1.*.fastq")
```

Comment:
add a bash for loop

for file in `ls ~{batch_id}_split_r1.*.fastq` ; do gzip $file ; 
done

Array[File] split_read1 = glob("~{batch_id}_split_r1.*.fastq.gz")

Reply:
From my test data, and it will actually double the time for this task. 

My thought is that, we might be able to cut disk usage by clicking on "Delete intermediate outputs" when running this WDL.